frontend/module/editor.js

```diff
@@ -343,25 +346,68 @@ const preserveIndent = () => (textareaProps, event) => {
 
   const lines = value.split("\n")
   const currentLine = lines[currentLineNumber]
-  const matches = /^\s+/.exec(currentLine)
-
-  if (!matches) {
+  const matches = /^(\s+)?((func|on|if|while|for)\b)?(end|else\b)?/.exec(currentLine)
```

Comment:
the `end|else` regexp is not quite right as the `\b` is only on the `else`. Needs parens like the other.

Reply:
whoops. I <3 regexp

Reply:
thank you.

Reply:
done.